tonic/src/transport/service/connector.rs

```diff
@@ -78,20 +79,43 @@ where
         #[cfg(feature = "tls-roots-common")]
         let tls = self.tls_or_default(uri.scheme_str(), uri.host());
 
+        let is_https = uri.scheme_str() == Some("https");
         let connect = self.inner.make_connection(uri);
 
         Box::pin(async move {
+            #[cfg(not(feature = "tls"))]
+            {
+                if is_https {
+                    return Err(NoTlsSupport(()).into());
+                }
+            }
+
             let io = connect.await?;
 
             #[cfg(feature = "tls")]
             {
                 if let Some(tls) = tls {
                     let conn = tls.connect(io).await?;
                     return Ok(BoxedIo::new(conn));
+                } else if is_https {
+                    return Err(NoTlsSupport(()).into());
                 }
             }
 
             Ok(BoxedIo::new(io))
         })
     }
 }
+
+/// Error returned when trying to connect to an HTTPS endpoint without TLS enabled.
+#[derive(Debug)]
+pub(crate) struct NoTlsSupport(());
```

Comment:
Probably should use a name like this https://github.com/hyperium/hyper-tls/blob/master/src/client.rs#L125 bit more informative than NoTlsSupport imo

Reply:
Changed the name to `HttpsUriWithoutTlsSupport`.